### ISSUE TITLE
Command history to work forgotten between sessions

### ISSUE THREAD

**Issue by tkittel** [prompt-toolkit/ptpython#357]
Hi there.

With ptpython 3.0.2 I can't get ptpython to remember the command history between sessions. I think it used to work on my old machine with an older version of ptpython. I would imagine that after using ptpython I would get the history stored in either ~/.ptpython/history or ~/.config/ptpython/history, but no such luck. I am running out of a venv using Python 3.6.9 on Ubuntu 18.04, and I launch ptpython in all Python sessions by having PYTHONSTARTUP point to a file with the following contents:
```python
import sys
print('Welcome')
from ptpython.repl import embed as _ptpython_embed
sys.exit(_ptpython_embed(globals(), locals())
```
Am I missing something obvious for the history to be made persistent between sessions?

**Comment by hemanta212** [comment]
Yes. Not working for me too. The history file never gets created.

**Comment by jnns** [comment]
Do you have a persisting history if you invoke `ptpython` directly? It works for me at least.

It seems that the `embed()` in `PYTHONSTARTUP` is missing a reference to the history file's location (which is obtained when invoking the REPL directly). I made it work with this snippet:

```
ptpython_dir = Path.home() / ".config" / "ptpython"
sys.path.insert(0, str(ptpython_dir))

try:
    from ptpython.repl import embed
except ImportError:
    print("ptpython is not available: falling back to standard prompt")
else:
    sys.exit(
        embed(
            history_filename=ptpython_dir / "history",
            globals=globals(),
            locals=locals(),
        )
    )
```

**Comment by tkittel** [comment]
Thanks @jnns !

The `history_filename` argument to `embed` did the trick for me. It also works without the `sys.path.insert` hack it seems. Or do you think it might be needed on some platforms?
